core/src/main/java/com/google/cloud/sql/core/ConnectionInfoCache.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+/** ConnectionInfoCache is the contract for a caching strategy for ConnectionInfo. */
+public interface ConnectionInfoCache {
+
+  /**
+   * Returns metadata needed to create a connection to the instance.
+   *
+   * @return returns ConnectionMetadata containing the preferred IP and SSL connection data.
+   * @throws IllegalArgumentException If the instance has no IP addresses matching the provided
+   *     preferences.
+   */
+  ConnectionMetadata getConnectionMetadata(long timeoutMs);
+
+  void forceRefresh();
+
+  void refreshIfExpired();
+
+  RefreshStrategy getRefresher();
+
+  CloudSqlInstanceName getInstanceName();
```

Comment:
This is used only in test. Let's remove the method here and either drop the usages in test or find a more direct way at getting at the information the test wants.

Reply:
I removed this method too.